api/metadata/metadata.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2021 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metadata
+
+import (
+	"context"
+	"sync"
+
+	"github.com/gravitational/teleport/api/constants"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+const versionKey = "version"
+
+var (
+	versionValue = constants.Version
+	disabled     = false
+	mu           sync.RWMutex
+)
+
+// SetVersion is used only for tests to set the version value that the client
+// will send in the GRPC metadata.
+func SetVersion(version string) {
+	mu.Lock()
+	defer mu.Unlock()
+	versionValue = version
+}
+
+// DisabledIs disables (or enables) including metadata in GRPC requests. Used in tests.
+func DisabledIs(metadataDisabled bool) {
+	mu.Lock()
+	defer mu.Unlock()
+	disabled = metadataDisabled
+}
+
+func addMetadataToContext(ctx context.Context) context.Context {
+	mu.RLock()
+	defer mu.RUnlock()
+	if disabled {
+		return ctx
+	}
+	return metadata.AppendToOutgoingContext(ctx, versionKey, versionValue)
+}
```

Comment:
I'm wondering if these test helper functions can be removed. Instead of operating on global state we can just set context ongoing  metadata  in the test like:

```go
ctx := context.Background()
if  tc.versionValue != "" {
    ctx  = metadata.AppendToOutgoingContext(ctx, versionKey, tc.versionValue)
}
gotRoles, err := client.GetRoles(ctx)
...
```

Reply:
Thanks for this comment, I hadn't considered that the caller can pass their own metadata through the client, I've updated the metadata package and the test to avoid using globals in the latest commit